### PR TITLE
EdkRepo: Create unit tests for the class ValidateManifest and modular…

### PR DESCRIPTION
**Pull request by VictorMlVillarreal**
…ize its existing methods if needed

Implemented unit tests and documentation for the ValidateManifest class to improve clarity, reliability, and maintainability.

Changes:
-Added unit tests for the ValidateManifest class, covering the main execution paths and edge cases. 
-Created the corresponding documentation file ValidateManifest_TestCases.md. 
-All new tests rely on mocked dependencies and do not perform filesystem or network operations.

Fixes: #325

### CODE REVIEW

**Review by ashedesimone — CHANGES_REQUESTED** [review]

File: edkrepo_manifest_parser/unit_tests/test_validatemanifest.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+ID_DIFF_CASE             = 'different_case'
+ID_UNEQUAL               = 'unequal'
+SINGLE_MATCH_FAIL_FIELDS = 'project,project_list,expected_msg'
+ID_NO_MATCH              = 'no_match'
+ID_MULTIPLE_MATCHES      = 'multiple_matches'
+
+
+class TestValidateManifest:
+
+    @pytest.fixture
```
Comment (ashedesimone) [inline edkrepo_manifest_parser/unit_tests/test_validatemanifest.py]: There is some added complexity being introduced by having one fixture depend on another which makes following the code flow more difficult. We can reduce this and further simplify our test case implementation by unifying the fixtures `mock_manifest_xml` and `mock_manifest_xml_instance` like so:

```
@pytest.fixture
def mock_manifest_xml(self):
    """Patch ManifestXml; yields a configured mock that returns a MagicMock instance."""
    with patch('edkrepo_manifest_parser.edk_manifest_validation.ManifestXml') as mock_cls:
        instance = MagicMock()
        mock_cls.return_value = instance
        # Yield both the class and instance for flexibility
        mock_cls.instance = instance
        yield mock_cls
```
and then updating the tests accordingly:

```
def test_validate_parsing_success(self, vm, mock_manifest_xml):
    """When ManifestXml succeeds, must return ("PARSING", True, None)."""
    result = vm.validate_parsing()
    assert result == ('PARSING', True, None)
    assert vm._manifest_xmldata is mock_manifest_xml.instance
```

**Review by ashedesimone — CHANGES_REQUESTED** [review]
